### DISCUSSION TITLE
S3 IAM PutUserAction behaviour

### DISCUSSION THREAD

**Discussion by vanhtuan0409** [seaweedfs/seaweedfs#3376] [General]
Currently, the behavior of `PutUserAction` is to append into an identity's actions list. I think this behavior does not match AWS behavior.
For example, `PutUserAction` cannot be used if we want to remove some specific actions from a user as it is an append-only API. To remove an action from a user, currently, I need to use `s3.configure` which will modify the `identities.json` directly

To solve this, I think `PutUserAction` behavior should replace the identity's action list. But this will be a breaking change. If you think is is acceptable, I am willing to put a PR for this

FYI: terraform provider using `PutUserAction` for both Create and Update action. Which I think imply that it should be a replacement behavior
https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/iam/user_policy.go#L22

**Comment by kmlebedev** [comment]
Let's burn.

**Comment by chrislusf** [comment]
The API is not on the data path. We can change the behavior with proper release announcement.
